File: 2026_GFG_POTD/JANUARY/02-01-2026/Sort 0s, 1s and 2s.js
/**
 * @param {number[]} arr
 * @returns {void}
 */

class Solution {
    sort012(arr) {
        let low = 0, mid = 0, high = arr.length - 1;
        while (mid <= high) {
            if (arr[mid] === 0) {
                [arr[low], arr[mid]] = [arr[mid], arr[low]];
                low++;
                mid++;
            } else if (arr[mid] === 1) {
                mid++;
            } else {
                [arr[mid], arr[high]] = [arr[high], arr[mid]];
                high--;
            }
        }
    }
}

File: 2026_GFG_POTD/JANUARY/02-01-2026/Sort 0s, 1s and 2s.c
void sort012(int arr[], int n) {
    int low = 0, mid = 0, high = n - 1;
    while (mid <= high) {
        if (arr[mid] == 0) {
            int t = arr[low];
            arr[low++] = arr[mid];
            arr[mid++] = t;
        } else if (arr[mid] == 1) {
            mid++;
        } else {
            int t = arr[mid];
            arr[mid] = arr[high];
            arr[high--] = t;
        }
    }
}